File: frontend/NewsHub/NewsHub/Models/Article.swift
import Foundation

struct Article: Codable, Identifiable {
    let id: String
    let title: String
    let description: String?
    let content: String?
    let author: String?
    let url: String
    let imageUrl: String?
    let sourceName: String
    let publishedAt: String
    let category: String
    let region: String?
    let sentimentScore: Double
    let sentimentLabel: String
    let relevanceScore: Double
    let aiSummary: String?
    let keyPoints: [String]
    let topicId: String?
    let createdAt: String
    
    enum CodingKeys: String, CodingKey {
        case id, title, description, content, author, url, category, region
        case imageUrl = "image_url"
        case sourceName = "source_name"
        case publishedAt = "published_at"
        case sentimentScore = "sentiment_score"
        case sentimentLabel = "sentiment_label"
        case relevanceScore = "relevance_score"
        case aiSummary = "ai_summary"
        case keyPoints = "key_points"
        case topicId = "topic_id"
        case createdAt = "created_at"
    }
}

File: backend/app/schemas/article.py
from typing import Optional, List
from pydantic import BaseModel
from datetime import datetime

class ArticleBase(BaseModel):
    title: str
    description: Optional[str] = None
    content: Optional[str] = None
    author: Optional[str] = None
    url: str
    image_url: Optional[str] = None
    source_name: str
    published_at: str
    category: str
    region: Optional[str] = None

class ArticleCreate(ArticleBase):
    pass

class ArticleUpdate(ArticleBase):
    pass

class ArticleResponse(ArticleBase):
    id: str
    sentiment_score: float
    sentiment_label: str
    relevance_score: float
    ai_summary: Optional[str] = None
    key_points: List[str] = []
    topic_id: Optional[str] = None
    created_at: datetime
    
    class Config:
        from_attributes = True

class FeedResponse(BaseModel):
    items: List[ArticleResponse]
    total: int
    page: int
    size: int
    pages: int
